Mathlib/CategoryTheory/Localization/Prod.lean

```diff
@@ -0,0 +1,132 @@
+
+namespace StrictUniversalPropertyFixedTarget
+
+variable {E : Type u₅} [Category.{v₅} E]
+  (F : C₁ × C₂ ⥤ E) (hF : (W₁.prod W₂).IsInvertedBy F)
+
+/-- Auxiliary definition for `prodLift`. -/
+noncomputable def prodLift₁ :
+    W₁.Localization ⥤ C₂ ⥤ E :=
+  Construction.lift (curry.obj F) (fun _ _ f₁ hf₁ => by
+    haveI : ∀ (X₂ : C₂), IsIso (((curry.obj F).map f₁).app X₂) :=
+      fun X₂ => hF _ ⟨hf₁, MorphismProperty.id_mem _ _⟩
+    apply NatIso.isIso_of_isIso_app)
+
+lemma prod_fac₁ :
+    W₁.Q ⋙ prodLift₁ F hF = curry.obj F :=
+  Construction.fac _ _
+
+/-- The lifting of a functor `F : C₁ × C₂ ⥤ E` inverting `W₁.prod W₂` to a functor
+`W₁.Localization × W₂.Localization ⥤ E` -/
+noncomputable def prodLift :
+    W₁.Localization × W₂.Localization ⥤ E := by
+  refine' uncurry.obj (Construction.lift (prodLift₁ F hF).flip _).flip
+  intro _ _ f₂ hf₂
+  haveI : ∀ (X₁ : W₁.Localization),
+      IsIso (((Functor.flip (prodLift₁ F hF)).map f₂).app X₁) := fun X₁ => by
+    obtain ⟨X₁, rfl⟩ := (Construction.objEquiv W₁).surjective X₁
+    exact ((MorphismProperty.RespectsIso.isomorphisms E).arrow_mk_iso_iff
+      (((Functor.mapArrowFunctor _ _).mapIso
+        (eqToIso (Functor.congr_obj (prod_fac₁ F hF) X₁))).app (Arrow.mk f₂))).2
+          (hF _ ⟨MorphismProperty.id_mem _ _, hf₂⟩)
+  apply NatIso.isIso_of_isIso_app
+
+lemma prod_fac₂ :
+    W₂.Q ⋙ (curry.obj (prodLift F hF)).flip = (prodLift₁ F hF).flip := by
+  simp only [prodLift, Functor.curry_obj_uncurry_obj, Functor.flip_flip]
+  apply Construction.fac
+
+lemma prod_fac :
+    (W₁.Q.prod W₂.Q) ⋙ prodLift F hF = F := by
+  rw [← Functor.uncurry_obj_curry_obj_flip_flip', prod_fac₂, Functor.flip_flip, prod_fac₁,
+    Functor.uncurry_obj_curry_obj]
+
+lemma prod_uniq (F₁ F₂ : (W₁.Localization × W₂.Localization ⥤ E))
+    (h : (W₁.Q.prod W₂.Q) ⋙ F₁ = (W₁.Q.prod W₂.Q) ⋙ F₂) :
+      F₁ = F₂ := by
+  apply Functor.curry_obj_injective
+  apply Construction.uniq
+  apply Functor.flip_injective
+  apply Construction.uniq
+  apply Functor.flip_injective
+  apply Functor.uncurry_obj_injective
+  simpa only [Functor.uncurry_obj_curry_obj_flip_flip] using h
+
+variable (W₁ W₂)
+
+/-- The universal property of the localized category for products of two of the
+constructed localized categories. -/
+noncomputable def prod :
+    StrictUniversalPropertyFixedTarget (W₁.Q.prod W₂.Q) (W₁.prod W₂) E where
+  inverts := (Localization.inverts W₁.Q W₁).prod (Localization.inverts W₂.Q W₂)
+  lift := prodLift
+  fac := prod_fac
+  uniq := prod_uniq
+
+end StrictUniversalPropertyFixedTarget
+
+variable (W₁ W₂)
+
+lemma Construction.prodIsLocalization :
+    (W₁.Q.prod W₂.Q).IsLocalization (W₁.prod W₂) :=
+  Functor.IsLocalization.mk' _ _
+    (StrictUniversalPropertyFixedTarget.prod W₁ W₂)
+    (StrictUniversalPropertyFixedTarget.prod W₁ W₂)
+
+end Localization
+
+open Localization
+
+namespace Functor
+
+namespace IsLocalization
+
+variable (W₁ W₂)
+
+instance prod [L₁.IsLocalization W₁] [L₂.IsLocalization W₂] :
```

Comment:
Maybe add a docstring here since this is the main result of the file?

Reply:
Thanks. I have added a docstring.